lang_GENERIC/analyze/Naming_AST.ml

```diff
@@ -254,6 +254,30 @@ let lookup_nonlocal_scope id scopes =
       None
 (*e: function [[Naming_AST.lookup_nonlocal_scope]] *)
 
+let make_type type_string =
+  Some(TyName(((type_string, Parse_info.fake_info "()"), {name_qualifier=None; name_typeargs=None})))
+
+let get_resolved_type (vinit, vtype) =
+  match vtype with
+    | Some(_) -> vtype
+    | None -> (
+      match vinit with
+         | Some(L (Bool _)) -> make_type "bool"
+         | Some(L (Int _)) -> make_type "int"
+         | Some(L (Float _)) -> make_type "float"
+         | Some(L (Char _)) -> make_type "char"
+         | Some(L (String _)) -> make_type "str"
+         | Some(L (Regexp _)) -> make_type "regexp"
+         | Some(L (Unit _)) -> make_type "unit"
+         | Some(L (Null _)) -> make_type "null"
+         | Some(L (Imag _)) -> make_type "imag"
+         | Some(Id (_, {id_type; _}) as exp) -> (
+          let s = Meta_AST.vof_expr (exp) |> OCaml.string_of_v in
```

Comment:
BTW those Meta_AST.vof_xxx would soon disappear.
You can now use AST_generic.show_expr exp
(those functions are now autogenerated on the fly based on the [@@deriving show] attribute in the AST generic definitions.